website/blog/2021-11-14-Presto-Cluster-Memory-Tuning.md

```diff
@@ -0,0 +1,70 @@
+---
+author: George Wang
+authorURL: https://www.linkedin.com/in/george-wang-9a5a46190/
+title: Presto Cluster Memory Tuning
+---
+
+##Presto Memory Management
+
+Presto's memory management system is carefully designed to achieve high performing query engine in a distributed system. It defines the memory into two categories: user memory and system memory.
+* The user memory is related to user data. For example, reading user data from cloud storage on AWS S3 bucket will occupy the corresponding memory. The amount of data is tightly related;
+* The system memory is meant for some memory allocated for the operator itself that needs to execute in order to process the query, such as tablescan operation and aggregation operation, for which the memory space may be required by operators, and are not strongly related to the user's data itself.
```

Comment:
aggregation is a funny case, because for partial aggregation we use system memory, but for final aggregation we use user memory.  System memory can be pretty strongly related to user data and not merely constant overhead for operators, but Presto doesn't really expect the user to be able to reason about it.  FWIW I feel this is a confusing and not especially useful distinction and I created an issue to get rid of this distinction https://github.com/prestodb/presto/issues/17461.